python/whylogs/api/writer/whylabs.py

```diff
@@ -237,8 +271,47 @@ def _do_upload(
                 "Missing dataset ID. Specify it via WHYLABS_DEFAULT_DATASET_ID environment "
                 "variable or on your write method"
             )
-        if self._api_key is None:
-            raise EnvironmentError("Missing API key. Specify it via WHYLABS_API_KEY environment variable.")
+
+    def _do_get_feature_weights(self) -> FeatureWeightResponse:
+        """Get latest version for the feature weights for the specified dataset
+
+        Returns
+        -------
+        FeatureWeightResponse
+            Response of the GET request, with segmentWeights and metadata.
+        """
+        self._validate_org_and_dataset()
+        self._validate_api_key()
+
+        result = self._get_column_weights()
```

Comment:
I think it would be ok to translate the response type back into a `FeatureWeights` here. Better to encapsulate the whylabs-client integration and the use of the auto generated service api types so that when the generated classes change we only have to update the code in this file.

Reply:
Ok, we should probably tweak the code to include version info, then? I think that's an important information for the user to get.

Reply:
You could include a `metadata: Dict` here, its also interesting to be able to set this when writing but maybe that's not supported yet.

Reply:
added metadata: Dict to FeatureWeights